cmake/OpenCVGenConfig.cmake

```diff
@@ -128,8 +128,11 @@ endif()
 #  Part 3/3: ${BIN_DIR}/win-install/OpenCVConfig.cmake  -> For use within binary installers/packages
 # --------------------------------------------------------------------------------------------
 if(WIN32)
-  set(OpenCV_INCLUDE_DIRS_CONFIGCMAKE "\"\${OpenCV_CONFIG_PATH}/include\" \"\${OpenCV_CONFIG_PATH}/include/opencv\"")
-  set(OpenCV2_INCLUDE_DIRS_CONFIGCMAKE "\"\"")
+
+  if (CMAKE_HOST_SYSTEM_NAME MATCHES Windows)
+    set(OpenCV_INCLUDE_DIRS_CONFIGCMAKE "\"\${OpenCV_CONFIG_PATH}/include\" \"\${OpenCV_CONFIG_PATH}/include/opencv\"")
+    set(OpenCV2_INCLUDE_DIRS_CONFIGCMAKE "\"\"")
+  endif ()
```

Comment:
What are build cases require this condition?
Host OS - ? (Windows / MinGW / etc)
Target OS - ? (Android ?)

BTW, Values are got from [line 101](https://github.com/Itseez/opencv/pull/6269/files#diff-95a1135dd9166d702efe3795b131bb40L101)